include/nbl/video/IGPUObjectFromAssetConverter.h

```diff
@@ -294,37 +383,49 @@ auto IGPUObjectFromAssetConverter::create(const asset::ICPUBuffer** const _begin
 	const auto assetCount = std::distance(_begin, _end);
 	auto res = core::make_refctd_dynamic_array<created_gpu_object_array<asset::ICPUBuffer> >(assetCount);
 
-
     const uint64_t alignment =
         std::max<uint64_t>(
-            std::max<uint64_t>(m_driver->getRequiredTBOAlignment(), m_driver->getRequiredUBOAlignment()),
-            std::max<uint64_t>(m_driver->getRequiredSSBOAlignment(), _NBL_SIMD_ALIGNMENT)
+            std::max<uint64_t>(_params.limits.bufferViewAlignment, _params.limits.SSBOAlignment),
+            std::max<uint64_t>(_params.limits.UBOAlignment, _NBL_SIMD_ALIGNMENT)
         );
     
-
-    auto reqs = m_driver->getDeviceLocalGPUMemoryReqs();
+    auto reqs = _params.device->getDeviceLocalGPUMemoryReqs();
     reqs.vulkanReqs.alignment = alignment;
-    const uint64_t maxBufferSize = m_driver->getMaxBufferSize();
+    const uint64_t maxBufferSize = _params.limits.maxBufferSize;
     auto out = res->begin();
     auto firstInBlock = out;
     auto newBlock = [&]() -> auto
     {
         return core::LinearAddressAllocator<uint64_t>(nullptr, 0u, 0u, alignment, maxBufferSize);
     };
     auto addrAllctr = newBlock();
+
+    core::smart_refctd_ptr<IGPUCommandPool> pool = _params.device->createCommandPool(_params.transferQueue->getFamilyIndex(), IGPUCommandPool::ECF_RESET_COMMAND_BUFFER_BIT);
+    auto fence = _params.device->createFence(static_cast<IGPUFence::E_CREATE_FLAGS>(0));
+    core::smart_refctd_ptr<IGPUCommandBuffer> cmdbuf;
+    _params.device->createCommandBuffers(pool.get(), IGPUCommandBuffer::EL_PRIMARY, 1u, &cmdbuf);
+
+    cmdbuf->begin(IGPUCommandBuffer::EU_ONE_TIME_SUBMIT_BIT);
```

Comment:
I'd take an already begun, resettable commandbuffer from the outside as a parameter.

Just how the `updateBufferRangeViaStagingBuffer` does it

Reply:
hmm actually I'd take a command pool and a reference to a command buffer smartpointer instead, because you wont wait for the submit to finish in this case, so you cannot reset or drop the original commandbuffer until the submit is done.

So in that case if an "autosubmit" happens, then you need to put the submitted commandbuffer onto a zombie list together with a semaphore or a fence (enter DeferredEventHandler), create a new commandbuffer, swap the reference in the user parameter to the function.